tests/baselines/reference/for-of32.errors.txt

```diff
@@ -0,0 +1,7 @@
+tests/cases/conformance/es6/for-ofStatements/for-of32.ts(1,10): error TS7022: 'v' implicitly has type 'any' because it is does not have a type annotation and is referenced directly or indirectly in its own initializer.
+
+
+==== tests/cases/conformance/es6/for-ofStatements/for-of32.ts (1 errors) ====
+    for (var v of v) { }

```

Comment:
I'm reading the spec section 13.6.4.12, ForIn/OfHeadEvaluation, and for this `let v of v` case, it looks like v is in a temporal dead zone. I'm not very familiar with these, but I guess that means referencing v in that position would be an error. I may simply open a bug for this, but I want to double check what the compiler is supposed to do for a case like that. I did notice that in the case of `let v = v;`, we treat it as having type any, so at least this is consistent.